src/main/java/com/mycompany/myapp/domain/Reservation.java

```diff
@@ -0,0 +1,67 @@
+package com.mycompany.myapp.domain;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "pool")
+public class Reservation {
+
+    private static final long serialVersionUID = 1L;
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sequenceGenerator")
+    @SequenceGenerator(name = "sequenceGenerator")
+    private Long id;
+
+    @NotNull
+    @OneToOne
+    private User user;
+
+    @NotNull
+    @ManyToOne
+    @JoinColumn(name = "poolId")
+    private Pool pool;
+
+    @NotNull
+    @Column(unique = false, nullable = false)
+    private int count;
+
+    public Reservation(@NotNull User user, @NotNull Pool pool, @NotNull int count) {
```

Comment:
We need another, empty constructor for Jackson to deserialize entities

Reply:
ok